airbyte-db/lib/src/main/java/io/airbyte/db/mongodb/MongoDatabase.java

```diff
@@ -55,32 +55,46 @@ public void close() throws Exception {
   }
 
   public com.mongodb.client.MongoDatabase getDatabase() {
-    return mongoClient.getDatabase(databaseName);
+    return database;
+  }
+
+  public MongoIterable<String> getDatabaseNames() {
+    return mongoClient.listDatabaseNames();
   }
 
   public MongoIterable<String> getCollectionNames() {
-    return getDatabase().listCollectionNames();
+    return database.listCollectionNames();
   }
 
   public MongoCollection<Document> getCollection(String collectionName) {
-    return getDatabase().getCollection(collectionName)
+    return database.getCollection(collectionName)
         .withReadConcern(ReadConcern.MAJORITY);
   }
 
+  public MongoCollection<Document> getOrCreateNewCollection(String collectionName) {
+    var collectionNames = StreamSupport
+        .stream(database.listCollectionNames().spliterator(), false)
+        .collect(Collectors.toSet());
```

Comment:
Nitpick. We have a helper for this that is slightly simpler:

```java
Set<String> collectionNames = MoreIterators.toSet(database.listCollectionNames().iterator());
```

Reply:
Thanks, done.